tensorflow/c/eager/c_api.cc

```diff
@@ -247,11 +310,17 @@ void TFE_ContextOptionsSetAsync(TFE_ContextOptions* options,
                                 unsigned char enable) {
   options->async = enable;
 }
+
 void TFE_ContextOptionsSetDevicePlacementPolicy(
     TFE_ContextOptions* options, TFE_ContextDevicePlacementPolicy policy) {
   options->policy = policy;
 }
 
+void TFE_ContextOptionsSetInferInputAttrs(TFE_ContextOptions* options,
+                                          unsigned char enable) {
+  options->input_attrs_inference = enable;
```

Comment:
Can we remove this option and just unconditionally enable the inference?

Reply:
I personally have no objection to remove this option. I put it there to ensure that my changes won't have any impact (in reliability or performance) with existing clients that probably set those attributes explicitly right now. But inferring them anyway will still guarantee the backward compatibility of the current API, which is what matters. So if all reviewers agree (@iganichev already gave his +1), I'll make the changes.